src/pages/experiments/[experimentId]/plots-and-tables/dot-plot/index.jsx

```diff
@@ -71,75 +89,249 @@ const DotPlotPage = (props) => {
+  const getComparedConfig = (updatedConfig) => _.pick(
+    updatedConfig,
+    ['useMarkerGenes',
+      'nMarkerGenes',
+      'selectedGenes',
+      'selectedCellSet',
+      'selectedPoints'],
+  );
+
+  const hasGroupsToCompare = (baseCluster, filterCluster) => {
+    if (cellSetsLoading || !baseCluster || !filterCluster) return false;
+
+    // filterBy has the shape louvain/louvain-1
+    const [filterRootNode, filterKey] = filterCluster.split('/');
+
+    // If 'All" is chosen for the dropdown, there will always be representation from more than 1 group
+    if (filterRootNode === 'All') return true;
+
+    const filterClusterCellIds = Array.from(cellSetProperties[filterKey].cellIds);
+
+    const baseClusterParent = cellSetHierarcy.find((cellSet) => cellSet.key === baseCluster);
+    const baseClusterKeys = baseClusterParent.children.map((child) => child.key);
+    const baseClusterCellIds = baseClusterKeys.map((key) => cellSetProperties[key].cellIds);
+
+    // Check if there is at least 2 baseClusters with elements common to filterCluster
+    let numIntersections = 0;
+    return baseClusterCellIds.some((baseCellIds) => {
+      if (filterClusterCellIds.some((filterCellId) => baseCellIds.has(filterCellId))) {
+        numIntersections += 1;
+      }
+
+      return numIntersections === 2;
+    });
+  };
+
+  const hasGroups = useMemo(
+    () => hasGroupsToCompare(config?.selectedCellSet, config?.selectedPoints),
+    [config?.selectedCellSet, config?.selectedPoints, cellSetsLoading],
+  );
+
   useEffect(() => {
-    if (config && config.selectedGenes.length === 0 && !genesFetching && !highestDispersionGenes) {
-      const PROPERTIES = ['dispersions'];
-      const tableState = {
-        pagination: {
-          current: 1, pageSize: config.nMarkerGenes, showSizeChanger: true, total: 0,
-        },
-        geneNamesFilter: null,
-        sorter: { field: PROPERTIES[0], columnKey: PROPERTIES[0], order: 'descend' },
-      };
+    if (cellSetsLoading) return;
 
-      dispatch(loadPaginatedGeneProperties(experimentId, PROPERTIES, plotUuid, tableState));
+    // Marker genes calculation needs that the cellIds in groupBy (refer to fn definition)
+    // be represented by more than one groups in filterBy to enable comparison
+    // if (config?.useMarkerGenes && !hasGroupsToCompare(config.selectedCellSet, config.selectedPoints)) {
+    if (config?.useMarkerGenes && !hasGroups) {
+      setMoreThanTwoGroups(false);
+      return;
     }
-  }, [highestDispersionGenes, config, genesFetching]);
+
+    setMoreThanTwoGroups(true);
+
+    const currentComparedConfig = getComparedConfig(config);
+    if (config && !_.isEqual(previousComparedConfig.current, currentComparedConfig)) {
+      previousComparedConfig.current = currentComparedConfig;
+      dispatch(fetchPlotDataWork(experimentId, plotUuid, plotType));
+    }
+  }, [config, cellSetProperties]);
+
+  const loadInitialCustomGenes = () => {
```

Comment:
Other plots use this function such as the violin plot and embedding continuous, can you check if you can move this function and reuse it in those places?

Reply:
I don't want to do it in this PR because it's too big already. I can make another ticket for it. Sounds good?